packages/async-flow/src/log-store.js

```diff
@@ -28,16 +31,38 @@ const LogStoreI = M.interface('LogStore', {
  * A growable, replayable, sequence of `LogEntry`s.
  *
  * @param {Zone} zone
+ * @param {object} [optMetadata]
+ * @param {number} [optMetadata.generation]
  */
-export const prepareLogStore = zone => {
+export const prepareLogStore = (zone, optMetadata) => {
+  const { generation = 0 } = optMetadata || {};
+
+  /**
+   * A predicate to indicate whether the entry is normally visible to the
+   * LogStore user, or is a more internal entry that only is visible via the
+   * Unfiltered methods.
+   * @param {LogEntry} entry
+   */
+  const entryIsVisible = entry => entry[0] !== 'startGeneration';
+
+  /**
+   * Only prepend these entries as part of each LogStore's first pushEntry.
+   * @type {LogEntry[]}
+   */
+  const entriesPrependedToLogStoreFirstPush = harden([
+    ['startGeneration', generation],
+  ]);
+
   /**
    * @type {Ephemera<LogStore, {
-   *           index: number
-   *           replayDoneKit: PromiseKit<undefined>
+   *           filteredIndex: number;
+   *           index: number;
+   *           replayDoneKit: PromiseKit<undefined>;
    *        }>}
    */
   const tmp = makeEphemera(log => {
     const result = {
+      filteredIndex: 0,
       index: 0,
```

Comment:
Given your method naming convention `foo` vs `unfilteredFoo`, should these state variables be renamed to

```suggestion
      index: 0,
      unfilteredIndex: 0,
```
?

Reply:
I think they should, so I've renamed the state variables accordingly.